src/t_hash.c

```diff
@@ -1885,22 +1889,27 @@ static ExpireAction hashTypeActiveExpire(eItem _hashObj, void *ctx) {
     activeExpireCtx->fieldsToExpireQuota -= info.itemsExpired;
 
     /* If hash has no more fields to expire, remove it from HFE DB */
+    robj *key = createStringObject(keystr, sdslen(keystr));
     if (info.nextExpireTime == EB_EXPIRE_TIME_INVALID) {
         if (hashTypeLength(hashObj, 0) == 0) {
-            robj *key = createStringObject(keystr, sdslen(keystr));
             dbDelete(activeExpireCtx->db, key);
-            notifyKeyspaceEvent(NOTIFY_GENERIC,"del",key, activeExpireCtx->db->id);
-            server.dirty++;
-            signalModifiedKey(NULL, &server.db[0], key);
```

Comment:
@moticless is `&server.db[0]` a mistake?

Reply:
That’s a bug. Good catch!